Sources/XiEditor/AppDelegate.swift

```diff
@@ -454,7 +454,7 @@ class AppDelegate: NSObject, NSApplicationDelegate, XiClient {
     // Provice a convenience helper for ordered documents
     private var orderedDocuments: [Document] {
         return NSApplication.shared.orderedDocuments
-            .compactMap { $0 as? Document }
+            .map { $0 as! Document }
```

Comment:
I think this is ok, it seems to me the documents should always be `Document`. Since @mmatoszko wrote this code, maybe he has an opinion?

Reply:
My feeling is that if this _isn't_ a `Document`, crashing is good.

Reply:
I'm not sure if document can crash here. Maybe going with `flatMap` would be a better option?

Reply:
@mmatoszko The only time this should crash is if we're somehow instantiating an `NSDocument` subclass that is not our own `Document`. To me that indicates a serious problem, and crashing would be good. Do you think there are any normal circumstances where this is possible?

Reply:
Ok, I get your point now, thanks 👍 
@SCdF could you drop a comment line, that the force unwrapping is intentional here?
We should reduce the amount of the force unwrappings in the future, and it would be nice to have an explicit information that we should omit this one.

Reply:
Also tiny nit, s/Provice/Provide/

Happy to approve after those minor changes.

Reply:
@mmatoszko I do also agree that we unwrap to much in general fwiw. 😄

Reply:
@liamrosenfeld great idea! 👍 
Going with `// swiftlint:disable force_unwrapping` would be super nice for this particular case..
I'm just not sure if this won't be a lot of work in the beginning 😄
I think we should first go with #260 in order to have a working CI and automate as much work as possible ^^

Opinions?

Reply:
Also: my opinion on force unwrapping is that it is not bad _per se_, but that it should be used judiciously to enforce invariants. So if an RPC has a defined schema, force unwrapping when extracting items from the RPC's params is fine; if it crashes it means either core is sending bad data or the client is misunderstanding the schema, and crashing is good in both those cases.

I think there's a common problem with things like,

```swift
if let thing = optionalThingThatShouldNeverBeNil {
    thing.doStuff()
}
// ignore the nil case
```

In that the nil case isn't handled at all, it just silently falls through, which can lead to some pretty hard to debug behaviour.

Reply:
OK I've added the comment and fixed that existing spelling mistake as noted.

I'm going to leave the linting to someone else, because I don't actually _know_ Swift (or, while we're talking about it, Rust really).